app/controllers/nodes_controller.rb

```diff
@@ -55,12 +55,19 @@ def update
 
     respond_to do |format|
       @node_form = NodeForm.new(node_form_params)
+
       if @node_form.save
         format.html { redirect_to node_path(@node.parent_id), flash: { success: "Node updated" }}
         format.json { render json: @node, status: :ok }
       else
-        format.html { redirect_to edit_node_path(@node.id), flash: { danger: "Failed to update node" }}
-        format.json { render json: @node.errors, status: :unprocessable_entity }
+        format.html {
+          flash[:error] = []
```

Comment:
Is this necessary? It's a little odd having flash[:error] being an array. Perhaps convert the errors to a sentence e.g. `errors.full_messages.to_sentence` instead